### ISSUE TITLE
Create a new release

### ISSUE THREAD

**Issue by LegNeato** [graphql-rust/juniper#340]
It's about time to push out a new release. I want to automate a bit more of the process as well.

We probably need the book updating working too.

**Comment by theduke** [comment]
We have some breaking changes merged into master but I think they are so small that they don't really merit a version bump yet.

We should probably cherry-pick the non-breaking stuff into a `0.11` branch and release from there.

**Comment by theduke** [comment]
Well, actually, https://github.com/graphql-rust/juniper/pull/312 is really minor and `ScalarValue` is still there, only deprecated, so master should be fine.

**Comment by LegNeato** [comment]
I landed changes to release automation. I think we want https://github.com/graphql-rust/juniper/pull/353 and then we can do a release. We should do a minor (not patch) version bump as we increased the minimum rust version.

**Comment by theduke** [comment]
If we are doing a minor now we should definitely also include #333  since it provide a significant usability increase.

It does bump rust requirement to 1.33 but I think that's fine for a evolving library.

**Comment by LegNeato** [comment]
Hmmm, that is a huge change. I'd prefer to do a release with current state and then land that, have it sit in master for a bit, and then cut another one.

**Comment by theduke** [comment]
It's a big change but not a user visible one right away since everyone will still be on the old macro. Also I've converted one large and two smallish projects and all works fine.

**Comment by LegNeato** [comment]
I don't think we need to wait on the book here. Do you want to run a release or shall I?

**Comment by theduke** [comment]
Can do, but since you did all the work on the releasing tool, you should have the honors of breaking it in.

**Comment by theduke** [comment]
@LegNeato actually I'd love to get #355  in as well since it's breaking.

**Comment by LegNeato** [comment]
Ok, I'll do a release later tonight! Probably best for me to do it so if automation breaks I can fix it

**Comment by theduke** [comment]
@LegNeato please also check Gitter, I sent you a message with a nice surprise.

**Comment by LegNeato** [comment]
Juniper 0.12.0 is out!

Automation seemed to work well. `cargo-release` is pretty brittle, so I added some additional directions to `RELEASING.md` suggesting a clean checkout.

**Comment by theduke** [comment]
@LegNeato for future reference, what was brittle about it?

**Comment by LegNeato** [comment]
The cargo-release parts error out when pushing to git if your repo's git remote branch isn't in sync with upstream/master.

A clean clone with this repo as origin makes all those issues impossible, so that's what I added to the docs.
